### ISSUE TITLE
nn.Module doesn't work with roberta

### ISSUE THREAD

**Issue by Srj** [pytorch/fairseq#2304]
I have declared my sentiment classifier like below. But it doesn't train. It saturates to 52% accuracy on SST2 dataset no matter how many epochs i run.

```

class Model(nn.Module):
    def __init__(self):
        super(Model,self).__init__()
        self.base =torch.hub.load('pytorch/fairseq', 'roberta.large')
        self.l0 = nn.Linear(1024,512)
        self.l1 = nn.Linear(512,256)
        self.l2 = nn.Linear(256,1)
        self.d0 = nn.Dropout(0.5)
        self.d1 = nn.Dropout(0.5)
        self.d2 = nn.Dropout(0.5)
        nn.init.normal_(self.l0.weight,std=0.2)
        nn.init.normal_(self.l1.weight,std=0.2)
        nn.init.normal_(self.l2.weight,std=0.2)
        
        
    def forward(self,ids):
        out = self.base.extract_features(ids)
        out = out[:,0,:]
        x = self.d0(out)
        x = self.l0(x)
        x = F.leaky_relu(x)
        x = self.d1(x)
        x = self.l1(x)
        x = F.leaky_relu(x)
        x = self.d2(x)
        x = self.l2(x)
        return x
        
```

**Comment by lematt1991** [comment]
You should try following the [custom classification example](https://github.com/pytorch/fairseq/blob/master/examples/roberta/README.custom_classification.md).  There isn't enough context here for us to help.
